endorsement/dao/gws.py

```diff
@@ -71,12 +71,19 @@ def get_group_admin_emails(group_id):
                 'email': admin.name,
                 'display_name': admin.display_name})
         elif admin.is_group():
-            for member in gws.get_effective_members(admin.name):
-                if member.is_uwnetid():
-                    emails.append({
-                        'email': uw_email_address(member.name),
-                        'display_name': member.display_name})
-                elif member.is_group():
-                    emails += get_group_admin_emails(member.name)
+            emails += get_effective_member_emails(admin.name)
+
+    return emails
+
+
+def get_effective_member_emails(group_id):
+    emails = []
+    for member in gws.get_effective_members(group_id):
```

Comment:
Not having looked at the actual `gws.get_effective_members` function I thought the API already did what you're asking in returning the effective members (including recursion)

Reply:
true, the is_group test a couple lines down isn't necessary, but the function wraps the return member netid with uw_email_address() so it's more suitable for email...